kalite/control_panel/api_resources.py

```diff
@@ -245,6 +245,31 @@ def alter_list_data_to_serialize(self, request, to_be_serialized):
             attempt_logs = AttemptLog.objects.filter(user=user, exercise_id=bundle.data["exercise_id"], context_type__in=["playlist", "exercise"])
             bundle.data["timestamp_first"] = attempt_logs.count() and attempt_logs.aggregate(Min('timestamp'))['timestamp__min'] or None
             bundle.data["timestamp_last"] = attempt_logs.count() and attempt_logs.aggregate(Max('timestamp'))['timestamp__max'] or None
+            bundle.data["unit"] = 0
+            current_unit = get_current_unit_settings_value(user.facility.id)
+            if current_unit == 101:
+                bundle.data["unit"] = 101
+            else:
+                x = bundle.data["timestamp_first"]
+                if x:
+                    try:
+                        x = datetime.strptime(str(x), '%Y-%m-%d %H:%M:%S')
```

Comment:
(Given that this will be running for *every* exerciselog, efficiency is important)